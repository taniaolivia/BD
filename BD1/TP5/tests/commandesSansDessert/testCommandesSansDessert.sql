-- ============================================================================
-- Fichier          : testCommandesSansDessert.sql
-- Auteur           : Tania OLIVIA
-- Date             : 18 Novembre 2019
-- Base		        : Surgeles
-- Role             : Le test pour les commandes sans dessert
-- Fichier resultat : testCommandesSansDessert.out
-- ============================================================================

-- ============================================================================
-- Modification du format de la date pour se conformer aux donnees du programme de test
-- ============================================================================
alter session set NLS_DATE_FORMAT='dd/mm/yyyy';

-- ============================================================================
-- Suppression des nuplets de la base de donnees
-- NB : Le test est effectué sur une base vide, il est nécessaire 
--      de supprimer tous les nuplets de toutes les relations
-- ============================================================================
@ ../../utilitaires/vider_base_surgeles.sql

-- ============================================================================
-- Jeu de test pour le DOMAINE VALIDE 
-- NB : L'insertion de chaque nuplet doit faire l'objet d'un commentaire, 
-- notamment pour motiver le/les cas dans le(s)quel(s) il intervient.
-- ============================================================================

-- CAS 1
-- 1ère commande de Jean MARTIN
------- Le client de la commande
INSERT INTO CLIENT (IDCLI, NOMCLI, PRENOMCLI, TELCLI, MAILCLI, ADRCLI)
VALUES(1, 'MARTIN', 'Jean', '0505050505', 'martin@example.org', 'La Rochelle'); 

-------la commande passée par le client 1
INSERT INTO COMMANDE (IDCOM, IDCLI, DATECOM, DATERECEPPREVI)
VALUES(4, 1, TO_DATE('01/12/2013', 'DD/MM/YYYY'), TO_DATE('20/12/2013', 'DD/MM/YYYY'));

INSERT INTO PRODUIT (IDPRO, NOMPRO, TYPEPRO)
VALUES (2, 'B', 'POISSON');

INSERT INTO LIGNE_COMMANDE (IDLIGCOM, IDCOM, IDPRO, QTECOM, PRIXUNIT)
VALUES (6, 4, 2, '5', '4');



-- ============================================================================
-- Jeu de test pour le DOMAINE INVALIDE 
-- ============================================================================

-- CAS 2
-- 1ère commande de Jean MARTIN
------- Le client de la commande
-- Le client est Jean MARTIN de code 1, MEME client que pour la commande 1 de cas 1, donc on ne le recrée pas
-------la commande qui n'a passé par le client 1
INSERT INTO COMMANDE (IDCOM, IDCLI, DATECOM, DATERECEPPREVI)
VALUES(1, 1, TO_DATE('01/09/2013', 'DD/MM/YYYY'), TO_DATE('20/12/2013', 'DD/MM/YYYY'));

INSERT INTO PRODUIT (IDPRO, NOMPRO, TYPEPRO)
VALUES (1, 'A', 'DESSERT');

INSERT INTO LIGNE_COMMANDE (IDLIGCOM, IDCOM, IDPRO, QTECOM, PRIXUNIT)
VALUES (1, 1, 1, '5', '10');



-- 2ème commande de Jean MARTIN
------- Le client de la commande
-- Le client est Jean MARTIN de code 1, MEME client que pour la commande 1 de cas 1, donc on ne le recrée pas
-------la commande qui n'a passé par le client 1
-- L'identifiant du produit est 1, MEME commande que pour la commande 1, donc on ne le recrée pas

INSERT INTO COMMANDE (IDCOM, IDCLI, DATECOM, DATERECEPPREVI)
VALUES(2, 1, TO_DATE('01/10/2013', 'DD/MM/YYYY'), TO_DATE('20/12/2013', 'DD/MM/YYYY'));

INSERT INTO LIGNE_COMMANDE (IDLIGCOM, IDCOM, IDPRO, QTECOM, PRIXUNIT)
VALUES (2, 2, 1, '5', '4');



-- 3ème commande de Jean MARTIN
------- Le client de la commande
-- Le client est Jean MARTIN de code 1, MEME client que pour la commande 1 de cas 1, donc on ne le recrée pas
-------la commande qui n'a passé par le client 1
-- L'identifiant de commande est 2, MEME commande que pour la commande 2, donc on ne le recrée pas
-- L'identifiant du produit est 2, MEME commande que pour la commande 1 de cas 1, donc on ne le recrée pas

INSERT INTO LIGNE_COMMANDE (IDLIGCOM, IDCOM, IDPRO, QTECOM, PRIXUNIT)
VALUES (3, 2, 2, '5', '2');



-- 4ème commande de Jean MARTIN
------- Le client de la commande
-- Le client est Jean MARTIN de code 1, MEME client que pour la commande 1 de cas 1, donc on ne le recrée pas
-------la commande qui n'a passé par le client 1
-- L'identifiant du produit est 1, MEME commande que pour la commande 1, donc on ne le recrée pas
INSERT INTO COMMANDE (IDCOM, IDCLI, DATECOM, DATERECEPPREVI)
VALUES(3, 1, TO_DATE('01/11/2013', 'DD/MM/YYYY'), TO_DATE('20/12/2013', 'DD/MM/YYYY'));

INSERT INTO LIGNE_COMMANDE (IDLIGCOM, IDCOM, IDPRO, QTECOM, PRIXUNIT)
VALUES (4, 3, 1, '5', '4');



-- 5ème commande de Jean MARTIN
------- Le client de la commande
-- Le client est Jean MARTIN de code 1, MEME client que pour la commande 1 de cas 1, donc on ne le recrée pas
-------la commande qui n'a passé par le client 1
-- L'identifiant de commande est 3, MEME commande que pour la commande 4, donc on ne le recrée pas
INSERT INTO PRODUIT (IDPRO, NOMPRO, TYPEPRO)
VALUES (3, 'C', 'DESSERT');

INSERT INTO LIGNE_COMMANDE (IDLIGCOM, IDCOM, IDPRO, QTECOM, PRIXUNIT)
VALUES (5, 3, 3, '5', '10');


-- ============================================================================
-- resultats a mettre dans le fichier a nommer obligatoirement:
-- 
-- ce fichier doit contenir:	
--	1. l'etat de la base avant execution de la requete (donc juste apres 
--         les INSERT)
--	2. le texte SQL de la requete
--	3. le resultat de l'execution de la requete
-- ============================================================================

-- Debut de l'ecriture du fichier resultat du programme de test 
SPOOL testCommandesSansDessert.out
PROMPT fichier resultat du test : testCommandesSansDessert.out

-- Appel du fichier de presentation
@ ../../utilitaires/presentation_surgeles.sql

-- Afficher le contenu des relations de la base de donnees
PROMPT Contenu de la base : 
@ ../../utilitaires/afficher_base_surgeles.sql

-- ============================================================================
-- Requete a tester 
-- ============================================================================

-- Debut de l'affichage du resultat de la requete sur le client SqlPlus
PROMPT 'Resultat de la requete avec IN'
SET echo ON
@commandesSansDessertIN.sql
SET echo OFF

PROMPT 'Resultat de la requete avec Exists'
SET echo ON
@commandesSansDessertExists.sql
SET echo OFF

prompt 'Resultat de la requete sans sous-requete'
SET echo ON
@commandesSansDessertSansSR.sql
SET echo OFF

-- Fin de l'ecriture du fichier resultat du programme de test 
SPOOL OFF

-- ============================================================================
-- Fin du programme de test
-- ============================================================================

