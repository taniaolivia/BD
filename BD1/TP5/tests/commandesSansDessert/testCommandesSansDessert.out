fichier resultat du test : testCommandesSansDessert.out
Contenu de la base :
Contenu de la table Client :
=================================

     IDCLI NOMCLI     PRENOMCLI      TELCLI MAILCLI            ADRCLI                                                                                 
---------- ---------- ---------- ---------- ------------------ ------------                                                                           
         1 MARTIN     Jean       ########## martin@example.org La Rochelle                                                                            

Contenu de la table Commande:
=================================

     IDCOM      IDCLI DATECOM    DATERECEPP                                                                                                           
---------- ---------- ---------- ----------                                                                                                           
         4          1 01/12/2013 20/12/2013                                                                                                           
         1          1 01/09/2013 20/12/2013                                                                                                           
         2          1 01/10/2013 20/12/2013                                                                                                           
         3          1 01/11/2013 20/12/2013                                                                                                           

Contenu de la table Ligne_Commande:
=================================

  IDLIGCOM      IDCOM      IDPRO     QTECOM   PRIXUNIT                                                                                                
---------- ---------- ---------- ---------- ----------                                                                                                
         6          4          2          5          4                                                                                                
         1          1          1          5         10                                                                                                
         2          2          1          5          4                                                                                                
         3          2          2          5          2                                                                                                
         4          3          1          5          4                                                                                                
         5          3          3          5         10                                                                                                

6 lignes sélectionnées.

Contenu de la table Produit:
=================================

     IDPRO NOMPRO     TYPEPRO                                                                                                                         
---------- ---------- ---------------------                                                                                                           
         2 B          POISSON                                                                                                                         
         1 A          DESSERT                                                                                                                         
         3 C          DESSERT                                                                                                                         

Contenu de la table Livraison:
=================================

aucune ligne sélectionnée

Contenu de la table Livreur:
=================================

aucune ligne sélectionnée

'Resultat de la requete avec IN'
SQL> @commandesSansDessertIN.sql
SQL> -- ============================================================================
SQL> -- Fichier 	 : commandesSansDessertIN.sql
SQL> -- Auteur		 : Tania OLIVIA
SQL> -- Date		 : 18 Novembre 2019
SQL> -- Base			     : Surgeles
SQL> -- Role		 : La requête avec IN pour connaître les numéros et date de commande qui ne comportant pas de produit de type "dessert"
SQL> -- ============================================================================
SQL> 
SQL> SELECT C.IDCOM, C.DATECOM
  2  FROM COMMANDE C
  3  WHERE C.IDCOM NOT IN (SELECT L.IDCOM
  4  		   FROM LIGNE_COMMANDE L, PRODUIT P
  5  		   WHERE L.IDPRO = P.IDPRO
  6  		   AND P.TYPEPRO = 'DESSERT'
  7  );

     IDCOM DATECOM                                                                                                                                    
---------- ----------                                                                                                                                 
         4 01/12/2013                                                                                                                                 

SQL> SET echo OFF
'Resultat de la requete avec Exists'
SQL> @commandesSansDessertExists.sql
SQL> -- ============================================================================
SQL> -- Fichier 	 : commandesSansDessertExists.sql
SQL> -- Auteur		 : Tania OLIVIA
SQL> -- Date		 : 18 Novembre 2019
SQL> -- Base			     : Surgeles
SQL> -- Role		 :  La requête avec EXISTS pour connaître les numéros et date de commande qui ne comportant pas de produit de type "dessert"
SQL> -- ============================================================================
SQL> 
SQL> SELECT C.IDCOM, C.DATECOM
  2  FROM COMMANDE C
  3  WHERE NOT EXISTS (SELECT *
  4  		   FROM LIGNE_COMMANDE L, PRODUIT P
  5  		   WHERE L.IDPRO = P.IDPRO
  6  		   AND P.TYPEPRO = 'DESSERT'
  7  		   AND C.IDCOM = L.IDCOM
  8  );

     IDCOM DATECOM                                                                                                                                    
---------- ----------                                                                                                                                 
         4 01/12/2013                                                                                                                                 

SQL> SET echo OFF
'Resultat de la requete sans sous-requete'
SQL> @commandesSansDessertSansSR.sql
SQL> -- ============================================================================
SQL> -- Fichier 	 : commandesAvecDessertSansSR.sql
SQL> -- Auteur		 : Tania OLIVIA
SQL> -- Date		 : 18 Novembre 2019
SQL> -- Base			     : Surgeles
SQL> -- Role		 : La requête sans sous-requête pour connaître les numéros et date de commande qui ne comportant pas de produit de type "dessert"
SQL> -- ============================================================================
SQL> 
SQL> SELECT C.IDCOM, C.DATECOM
  2  FROM COMMANDE C, LIGNE_COMMANDE L, PRODUIT P
  3  WHERE C.IDCOM = L.IDCOM
  4  AND L.IDPRO = P.IDPRO
  5  AND P.TYPEPRO != 'DESSERT';

     IDCOM DATECOM                                                                                                                                    
---------- ----------                                                                                                                                 
         2 01/10/2013                                                                                                                                 
         4 01/12/2013                                                                                                                                 

SQL> SET echo OFF
